test/Riok.Mapperly.Tests/Mapping/ObjectNestedPropertyTest.cs

```diff
@@ -0,0 +1,261 @@
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapNestedProperties(nameof(A.Value))] partial B Map(A source);",
+            "class A { public string Id { get; set; } public C Value { get; set; } }",
+            "class B { public string Id { get; set; } }",
+            "class C { public string Id { get; set; } }"
+        );
+
+        TestHelper
+            .GenerateMapper(source, ignoreNestedMemberNotUsed)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                var target = new global::B();
+                target.Id = source.Id;
+                return target;
+                """
+            );
+    }
+
+    [Fact]
+    public void NestedPropertyWithSourcePathName()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapNestedProperties(nameof(A.Value))] partial B Map(A source);",
+            "class A { public C Value { get; set; } }",
+            "class B { public string ValueId { get; set; } }",
+            "class C { public string ValueId { get; set; } }"
+        );
+
+        TestHelper
+            .GenerateMapper(source)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                var target = new global::B();
+                target.ValueId = source.Value.ValueId;
+                return target;
+                """
+            );
+    }
+
+    [Fact]
+    public void NestedPropertyWithSourcePathNamePrefersAutoFlattenedCompletePath()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapNestedProperties(nameof(A.Value))] partial B Map(A source);",
+            "class A { public C Value { get; set; } }",
+            "class B { public string ValueId { get; set; } }",
+            "class C { public string Id { get; set; } public string ValueId { get; set; } }"
+        );
+
+        TestHelper
+            .GenerateMapper(source, ignoreNestedMemberNotUsed)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                var target = new global::B();
+                target.ValueId = source.Value.Id;
+                return target;
+                """
+            );
+    }
+
+    [Fact]
+    public void UnusedNestedPropertiesShouldDiagnostic()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapNestedProperties(nameof(A.Value))] partial B Map(A source);",
+            "class A { public string Id { get; set; } public C Value { get; set; } }",
+            "class B { public string Id { get; set; } }",
+            "class C { public string MyId { get; set; } }"
+        );
+
+        TestHelper
+            .GenerateMapper(source, TestHelperOptions.AllowDiagnostics)
+            .Should()
+            .HaveDiagnostic(DiagnosticDescriptors.NestedMemberNotUsed)
+            .HaveDiagnostic(DiagnosticDescriptors.SourceMemberNotMapped)
```

Comment:
Assert the generated diagnostics at least once with a snapshot to ensure the message template is rendered correctly and the location of the diagnostic is correct.